### PR TITLE
Feature sprint09 tijn multiselect download share

### PR DESCRIPTION
**Pull request by tschmits**

### CODE REVIEW

File: delft3dcontainermanager/tasks.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def get_docker_log(self, container_id, stdout=True, stderr=False, tail=5):
     """
     Retrieve the log of a container and return container id and log
     """
-    client = Client(base_url='http://localhost:4000')
+    client = Client(base_url='unix:///var/run/docker.sock')
```
Comment (evetion) [inline delft3dcontainermanager/tasks.py:78]: This won't break things right?

File: delft3dworker/models.py [hunk 2]
```diff
@@ -504,6 +495,9 @@ def publish_company(self, user):
         self.save()
 
     def publish_world(self, user):
+        if self.phase != self.phases.fin:
```
Comment (evetion) [inline delft3dworker/models.py:498]: Check if we're already published, just like publish_company

File: delft3dworker/views.py [hunk 3]
```diff
@@ -407,6 +410,22 @@ def publish_company(self, request, pk=None):
...
 
+    @list_route(methods=["post"])  # denied after publish to world
+    def publish_company_all(self, request):
+        queryset = Scene.objects.filter(owner=self.request.user).filter(
+                suid__in=request.data.getlist('suid', []))
+
+        try:
+            for scene in queryset:
+                scene.publish_company(request.user)
+        except ValueError, e:
```
Comment (evetion) [inline delft3dworker/views.py:421]: When does publish give us a ValueError? We already filtered existing ids with our queryset right?

So with a queryset [nicescene1, nicescene2, valueerrorscene, nicescene3], we get a 400 error, but two scenes are published, and one is not.

File: delft3dworker/views.py [hunk 4]
```diff
@@ -419,32 +438,98 @@ def publish_world(self, request, pk=None):
...
+            content_type="application/x-zip-compressed"
+        )
+        resp[
+            'Content-Disposition'] = 'attachment; filename={}'.format(
+                '{}.zip'.format(slugify(scene.name))
+        )
+
+        return resp
+
+    @list_route(methods=["get"])
```
Comment (evetion) [inline delft3dworker/views.py:496]: Can't there be one export function? Export always passes a list, but sometimes just with one item?

File: delft3dcontainermanager/tasks.py (hunk 1 above)
Comment (tschmits) [reply delft3dcontainermanager/tasks.py:78]: These values are updated by each provision:

- If the provision vars are correct, each provision will update this file correctly regardless of what is committed to this repository
- If I don't commit it, somebody else will in the future, I guess :)

File: delft3dworker/models.py (hunk 2 above)
Comment (tschmits) [reply delft3dworker/models.py:498]: Actually not: (as long as the Scene is in a finished state) the user can publish to world, no matter what state the run is in, i.e. "p", "c" or "w" (this last one is superfluous, but not incorrect - one can publish to world as often as one wishes).

File: delft3dworker/views.py (hunk 3 above)
Comment (tschmits) [reply delft3dworker/views.py:421]: https://docs.djangoproject.com/en/1.10/topics/db/queries/#querysets-are-lazy

Which means that the "`suid` correctness check" happens on r.419.

(The long read: https://docs.djangoproject.com/en/1.10/ref/models/querysets/#when-querysets-are-evaluated)

File: delft3dworker/views.py (hunk 4 above)
Comment (tschmits) [reply delft3dworker/views.py:496]: Correct. I intended this implementation to be a parallel changeover:

- making as little as possible changes on the front-end would mean less work
- backward compatibility (new BE works with old FE)
- lowering the chances of introducing a bug that would make downloading data completely impossible

I would suggest phasing out the old export function when the new one has proven it's worthy of existence (dramatic music).

**Pull request merged**